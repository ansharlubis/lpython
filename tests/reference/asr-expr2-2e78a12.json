{
    "basename": "asr-expr2-2e78a12",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/expr2.py",
    "infile_hash": "52d7d4d33553138f2cf55b9900047e5310c54d62e54b3ca1fa394024",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-expr2-2e78a12.stdout",
    "stdout_hash": "3f443139f08ef22841e5c9787c3253431f32a798cb05912370f36b14",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}