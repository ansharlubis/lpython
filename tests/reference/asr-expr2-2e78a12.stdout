([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      1 
      {
         test_boolOp: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  2 
                  {
                     a: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           a [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mLogical[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     b: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           b [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mLogical[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               test_boolOp [] [
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) 
                  ([1m[35mLogicalConstant[39m[0m .false. 
                     ([1m[35mLogical[39m[0m 4 [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mb[39m) 
                  ([1m[35mLogicalConstant[39m[0m .true. 
                     ([1m[35mLogical[39m[0m 4 [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) 
                  ([1m[35mLogicalBinOp[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ma[39m) [1m[32m
                     And[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mb[39m) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mb[39m) 
                  ([1m[35mLogicalBinOp[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ma[39m) [1m[32m
                     Or[39m[0m 
                     ([1m[35mLogicalConstant[39m[0m .true. 
                        ([1m[35mLogical[39m[0m 4 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) 
                  ([1m[35mLogicalBinOp[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ma[39m) [1m[32m
                     Or[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mb[39m) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) 
                  ([1m[35mLogicalBinOp[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ma[39m) [1m[32m
                     And[39m[0m 
                     ([1m[35mLogicalCompare[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33mb[39m) [1m[32m
                        Eq[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33mb[39m) 
                        ([1m[35mLogical[39m[0m 4 []) ()) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) 
                  ([1m[35mLogicalBinOp[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ma[39m) [1m[32m
                     And[39m[0m 
                     ([1m[35mLogicalCompare[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33mb[39m) [1m[32m
                        NotEq[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33mb[39m) 
                        ([1m[35mLogical[39m[0m 4 []) ()) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) 
                  ([1m[35mLogicalBinOp[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mb[39m) [1m[32m
                     Or[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mb[39m) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ())] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.)
      }) 
   [])
(TranslationUnit (SymbolTable 1 {main_program: (Program (SymbolTable 3 {}) main_program [] []), test_boolOp: (Subroutine (SymbolTable 2 {a: (Variable 2 a Local () () Default (Logical 4 []) Source Public Required .false.), b: (Variable 2 b Local () () Default (Logical 4 []) Source Public Required .false.)}) test_boolOp [] [(= (Var 2 a) (LogicalConstant .false. (Logical 4 [])) ()) (= (Var 2 b) (LogicalConstant .true. (Logical 4 [])) ()) (= (Var 2 a) (LogicalBinOp (Var 2 a) And (Var 2 b) (Logical 4 []) ()) ()) (= (Var 2 b) (LogicalBinOp (Var 2 a) Or (LogicalConstant .true. (Logical 4 [])) (Logical 4 []) ()) ()) (= (Var 2 a) (LogicalBinOp (Var 2 a) Or (Var 2 b) (Logical 4 []) ()) ()) (= (Var 2 a) (LogicalBinOp (Var 2 a) And (LogicalCompare (Var 2 b) Eq (Var 2 b) (Logical 4 []) ()) (Logical 4 []) ()) ()) (= (Var 2 a) (LogicalBinOp (Var 2 a) And (LogicalCompare (Var 2 b) NotEq (Var 2 b) (Logical 4 []) ()) (Logical 4 []) ()) ()) (= (Var 2 a) (LogicalBinOp (Var 2 b) Or (Var 2 b) (Logical 4 []) ()) ())] Source Public Implementation () .false. .false.)}) [])
